support/elasticsearch_search.erl

```diff
@@ -331,6 +331,11 @@ map_query(_, _) ->
 -spec map_must_not({atom(), any()}, z:context()) -> {true, list()} | false.
 map_must_not({cat_exclude, []}, _Context) ->
     false;
+
+%% ne undefined is translated into the filter "exists" {"field": Key} in 
+%% the map_filter function
+map_must_not({filter, [_Key, ne, undefined]}, _Context) ->
+    false;
```

Comment:
Alternatively, add an `undefined` guard to line 344.

Reply:
I don't think I understand? You mean this clause right:
```
 map_must_not({filter, [Key, Operator, Value]}, _Context) when Operator =:= '<>'; Operator =:= ne ->
{true, [{term, [{Key, z_convert:to_binary(Value)}]}]};
```
That returns a {true, ..., which will result in this filter popping up in the "must_not" part of the query, while we're looking for an "exists" in the "must" part of the query?

Reply:
What I mean is remove 

```erlang
map_must_not({filter, [_Key, ne, undefined]}, _Context) ->
     false;
```

and just before

```erlang
map_must_not({filter, [Key, Operator, Value]}, _Context) when Operator =:= '<>'; Operator =:= ne ->
```

insert

```erlang
map_must_not({filter, [Key, Operator, undefined]}, _Context) when Operator =:= '<>'; Operator =:= ne ->
```

This way, we cover both `ne` and `<>`, and it’s a bit clearer that these two clauses belong together.

Reply:
Ah! Yeah, that seems smart.